crates/biome_service/src/settings.rs

```diff
@@ -475,8 +475,12 @@ impl From<JsonConfiguration> for LanguageSettings<JsonLanguage> {
     fn from(json: JsonConfiguration) -> Self {
         let mut language_setting: LanguageSettings<JsonLanguage> = LanguageSettings::default();
 
-        language_setting.parser.allow_comments = json.parser.allow_comments;
-        language_setting.parser.allow_trailing_commas = json.parser.allow_trailing_commas;
+        if json.parser.allow_comments {
```

Comment:
Should we also refactor `JsonConfiguration`? If this has a default value, then all the subsequent works seems pointless.

Reply:
I don't think so, because afaik the `Partial` macro already does that. I'll take a closer look.

Reply:
I think `partial` is to derive a `PartialXXX` struct if it annotates a struct `XXX`. I can't see the reason to use the `partial` macro to a primitive value.

Also, the `JsonParser` itself does use `bool` type values. So in `From<JsonConfiguration>` we do only have `true` or  `false`.

I must say I'm also not very familiar with the infra. 🤷‍♂️

Reply:
When configuration is actually loaded, it's passed in as `PartialConfiguration`. For example:

https://github.com/biomejs/biome/blob/5fe2f68ae99068e950b2ec483fef89763b2e8db4/crates/biome_cli/src/commands/check.rs#L31

So I believe this concern is already handled. Otherwise I would think that a bunch more things would be broken.

Reply:
hmmm, let's focus on this part of the code, this is a `bool` right?

![image](https://github.com/biomejs/biome/assets/10386119/1848e33f-e7d1-4ea9-a1ac-e849550dfd0c)

If it is `false`, how do you determine if it is set by the user or it is the default value of this type?

The reason things weren't broken is because most of the commen options are indeed `Option<Type>`: https://github.com/biomejs/biome/blob/5fe2f68ae99068e950b2ec483fef89763b2e8db4/crates/biome_configuration/src/json.rs#L51-L74

We probably have some descrepencies across the codebase on using `Partial` or the original types.

Reply:
Ah, I see your point. This should be `impl From<PartialJsonConfiguration>`.